services-directions/src/test/java/com/mapbox/api/directions/v5/models/RouteOptionsTest.java

```diff
@@ -214,4 +215,106 @@ public void toJson_fromJson() throws Exception {
 
     assertEquals(routeOptions, routeOptionsFromJson);
   }
+
+  @Test
+  public void test_fromJson() {
+    String jsonString = "{" +
+      "\"profile\": \"auto\"," +
+      "\"user\": \"mapbox\"," +
+      "\"baseUrl\": \"https://api.mapbox.com\"," +
+      "\"coordinates\": [{" +
+      "\"type\": \"Point\"," +
+      "\"coordinates\": [-3.707788, 40.395039]" +
+      "}, {" +
+      "\"type\": \"Point\"," +
+      "\"coordinates\": [-3.712179, 40.401819]" +
+      "}]," +
+      "\"access_token\": \"ACCESS_TOKEN\"," +
+      "\"geometries\": \"polyline6\"," +
+      "\"overview\": \"full\"," +
+      "\"steps\": true," +
+      "\"bearings\": \";\"," +
+      "\"continue_straight\": true," +
+      "\"annotations\": \"congestion,distance\"," +
+      "\"language\": \"en\"," +
+      "\"roundabout_exits\": true," +
+      "\"voice_instructions\": true," +
+      "\"banner_instructions\": true," +
+      "\"voice_units\": \"imperial\"," +
+      "\"uuid\": \"uuid1\"" +
+      "}";
+
+    RouteOptions routeOptions = RouteOptions.fromJson(jsonString);
+    assertEquals("auto", routeOptions.profile());
+    assertEquals("mapbox", routeOptions.user());
+    assertEquals("https://api.mapbox.com", routeOptions.baseUrl());
+    assertEquals(2, routeOptions.coordinates().size());
+    assertEquals("ACCESS_TOKEN", routeOptions.accessToken());
+    assertEquals("polyline6", routeOptions.geometries());
+    assertEquals("full", routeOptions.overview());
+    assertEquals(true, routeOptions.steps());
+    assertEquals(";", routeOptions.bearings());
+    assertEquals(true, routeOptions.continueStraight());
+    assertEquals("congestion,distance", routeOptions.annotations());
+    assertEquals("en", routeOptions.language());
+    assertEquals(true, routeOptions.roundaboutExits());
+    assertEquals(true, routeOptions.voiceInstructions());
+    assertEquals(true, routeOptions.bannerInstructions());
+    assertEquals("imperial", routeOptions.voiceUnits());
+    assertEquals("uuid1", routeOptions.requestUuid());
+  }
+
+  @Test
+  public void test_toJson() {
+
+    RouteOptions routeOptions = RouteOptions.builder()
+      .profile("auto")
+      .user("mapbox")
+      .coordinates(Arrays.asList(Point.fromLngLat(-3.707788, 40.395039),
+        Point.fromLngLat(-3.712179, 40.401819)))
+      .accessToken("ACCESS_TOKEN")
+      .baseUrl("https://api.mapbox.com")
+      .geometries("polyline6")
+      .overview("full")
+      .steps(true)
+      .bearings(";")
+      .continueStraight(true)
+      .annotations("congestion,distance")
+      .language("en")
+      .roundaboutExits(true)
+      .voiceInstructions(true)
+      .bannerInstructions(true)
+      .voiceUnits("imperial")
+      .requestUuid("uuid1")
+      .build();
+
+    String jsonString = routeOptions.toString();
```

Comment:
@osana 
Found it 🎉 

Should be `routeOptions.toJson()` instead of `toString()`.

In any case, we should be careful because we can't compare json strings as we're doing here (`compareJson` below), because the order in which the json fields are represented is taken into account in the assertion which doesn't mean that the JSON is incorrect.

I've also noticed a discrepancy in the `coordinates` which we should fix 👀 

Expected
```
"coordinates":[-3.707788,40.395039]},{"type":"Point","coordinates":[-3.712179,40.401819]}]
```

Actual
```
"coordinates":[[-3.707788,40.395039],[-3.712179,40.401819]]
```